iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -195,6 +196,7 @@ void PortManager::handleSubscriberPorts() noexcept
 void PortManager::doDiscoveryForSubscriberPort(SubscriberPortType& subscriberPort) noexcept
 {
     subscriberPort.tryGetCaProMessage().and_then([this, &subscriberPort](auto caproMessage) {
+        cxx::Expects(caproMessage.m_serviceDescription.isValid() && "invalid service description!");
```

Comment:
pPlease do a check for a valid `ServiceDescription` in `RouDiprocessMessage`. It's not sufficient to do it only in the posh runtime since a user could open the socket and send a manipulated message (same could happen with fuzzing tests). We need to check all parameter for validity when they are de-serialized.

Reply:
Correct, that's important. Added.